src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -783,19 +783,35 @@ MulticopterPositionControl::start_flight_task()
 			// we want to be in this mode, reset the failure count
 			_task_failure_count = 0;
 		}
+	}
 
-	} else if (_control_mode.flag_control_auto_enabled) {
+	// Auto
+	const bool vehicle_in_auto_mode = (_vehicle_status.nav_state == vehicle_status_s::NAVIGATION_STATE_AUTO_RTGS
```

Comment:
What's this fixing? Is there a nav state that doesn't fit into control mode?

Reply:
It is just more consistent because all other switches are caused by the `vehicle_nav_state`